lib/effects/move_effect.dart

```diff
@@ -96,7 +96,11 @@ class MoveEffect extends SimplePositionComponentEffect {
     }
     final double totalPathLength = isAlternating ? pathLength * 2 : pathLength;
     speed ??= totalPathLength / duration;
+
+    /// `duration` is not null when speed is null
     duration ??= totalPathLength / speed;
+
+    /// `speed` is always not null here already
```

Comment:
since these are within the method body, I think they can be regular (`//`) comments (not dartdoc)